File: 122-best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        for price in range(1, len(prices)):
            if prices[price]-prices[price -1] > 0:
                profit = profit + prices[price]-prices[price -1]

        return profit

        

File: 122-best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int minPrice = prices[0];

        for(int i = 0;  i < prices.length; i++){
            if(prices[i] - minPrice > 0){

                maxProfit += prices[i] - minPrice;
                minPrice = prices[i];

                //minPrice = Integer.MAX_VALUE;
            }  else {
                 minPrice = Math.min(minPrice, prices[i]);
            }  
           // } else if(minPrice == Integer.MAX_VALUE){
           //     minPrice = prices[i];
           // }
        }

        return maxProfit;
    }
}